### ISSUE TITLE
Correct usage?

### ISSUE THREAD

**Issue by davidshumway** [adapt-python/adapt#62]
New to this area. Is this the correct usage? I'm not sure as it seems the target domain prediction appears to be nearly perfectly predicted. 

```python
from adapt.utils import make_classification_da
from adapt.feature_based import FA
Xs, ys = (x[x['ID'] == 1091], x[x['ID'] == 1091]['result'])
Xt, yt = (x[x['ID'] == 1437])], x[x['ID'] == 1437])]['result'])
model = FA(RandomForestRegressor(), Xt=Xt[0:200], yt=yt[0:200], random_state=0)
model.fit(Xs, ys)
model.score(Xt[0:200], yt[0:200])
```
> Fit transform...
Previous shape: (23564, 113)
New shape: (23764, 339)
Fit Estimator...
0.9998569213911123

```python
print(Xt.shape)
```
> (1797, 113)

```python
model.predict(Xt[200:220])[0:20]
```
> array([1.        , 1.        , 2.12669471, 1.79927106, 1.        ,
       1.        , 1.49136169, 1.        , 1.        , 1.98069632,
       1.30103   , 1.        , 1.92087984, 1.30103   , 1.79927106,
       1.92130166, 1.        , 1.30103   , 1.98065084, 1.7914019 ])

```python
yt[200:220]
```
> 223315   1.00
223330   1.00
225808   2.13
225812   1.80
225893   1.00
227047   1.00
227206   1.49
227230   1.00
227510   1.00
227578   1.98
227597   1.30
228051   1.00
228058   1.93
228117   1.30
228491   1.80
228549   1.93
228631   1.00
229026   1.30
229700   1.98
230579   1.79
Name: result, dtype: float64

```
mean_squared_error(yt[200:220], model.predict(Xt[200:220])[0:20], squared=False)
```
> 0.0044

**Comment by antoinedemathelin** [comment]
Hi @davidshumway ,
Thank you for your interest in Adapt!

It really depends on what you are trying to do, but obviously here you are overfitting `Xt`.

`FA` is a supervised domain adaptation method that use labeled data from both the source domain `Xs, ys` and the target domain `Xt, yt`, however the typical framework in this case is that you only have access to a few labels in the target domain. Here you give the full `Xt, yt` to `FA` which are then used to fit your `RandomForestRegressor`. If you want to consider a more realistic domain adaptation scenario, I suggest you try the following code:

```
from adapt.feature_based import FA
from sklearn.model_selection import train_test_split
Xs, ys = (X_train[X_train['ID'] == 1091], y_train[y_train['ID'] == 1091])
Xt, yt = (X_train[X_train['ID'] == 1437], y_train[y_train['ID'] == 1437])
train_index, test_index = train_test_split(np.arange(len(Xt)), train_size=0.1)

model = FA(RandomForestRegressor(), Xt=Xt[train_index], yt=yt[train_index], random_state=0)
model.fit(Xs, ys)
model.score(Xt[test_index], yt[test_index])
```

The smaller the `train_size` the harder it will be for the model to fit `Xt` and you will need the source data.
To identify the benefit of using `FA` in this case, you can compare the score with the results you get with the TargetOnly model:

```
tgt_only = RandomForestRegressor()
tgt_only.fit(Xt[train_index], yt[train_index])
tgt_only.score(Xt[test_index], yt[test_index])
```

**Comment by davidshumway** [comment]
Makes sense. Thanks, @antoinedemathelin!
